File: typescript/src/process1/starter.ts
import { Client, Connection } from '@temporalio/client';
import { randomUUID } from 'crypto';
import { registrationWorkflow } from './workflow';
import { TASK_QUEUE, TEMPORAL_URL, type RegisterRequest } from './models';

async function main() {
  const connection = await Connection.connect({ address: TEMPORAL_URL });
  const client = new Client({ connection });
  const userId = randomUUID();
  const request: RegisterRequest = {
    userId,
    email: 'flow@gmail.com',
  };

  const handle = await client.workflow.start(registrationWorkflow, {
    args: [request],
    taskQueue: TASK_QUEUE,
    workflowId: `registration-ts-${userId}`,
  });

  console.log(`WorkflowId: ${handle.workflowId}`);
  await connection.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python/src/temporal_learn/process1/starter.py
import asyncio
import uuid

from temporalio.client import Client

from .models import TASK_QUEUE, TEMPORAL_URL, RegisterRequest
from .workflow import RegistrationWorkflow


async def main():
    client = await Client.connect(TEMPORAL_URL)
    user_id = str(uuid.uuid7())
    request = RegisterRequest(user_id=user_id, email="flow@gmail.com")
    handle = await client.start_workflow(
        RegistrationWorkflow.register,
        request,
        id=f"registration-{user_id}",
        task_queue=TASK_QUEUE,
    )
    print(f"WorkflowId: {handle.id}")


if __name__ == "__main__":
    asyncio.run(main())
